nifi-extension-bundles/nifi-couchbase-bundle/nifi-couchbase-standard-services/src/main/java/org/apache/nifi/services/couchbase/StandardCouchbaseClient.java

```diff
@@ -0,0 +1,162 @@
+import com.couchbase.client.core.error.RequestCanceledException;
+import com.couchbase.client.core.error.ScopeNotFoundException;
+import com.couchbase.client.core.error.ServerOutOfMemoryException;
+import com.couchbase.client.core.error.ServiceNotAvailableException;
+import com.couchbase.client.core.error.TemporaryFailureException;
+import com.couchbase.client.core.error.ValueTooLargeException;
+import com.couchbase.client.core.error.subdoc.PathExistsException;
+import com.couchbase.client.core.error.subdoc.PathNotFoundException;
+import com.couchbase.client.java.Collection;
+import com.couchbase.client.java.codec.RawBinaryTranscoder;
+import com.couchbase.client.java.codec.RawJsonTranscoder;
+import com.couchbase.client.java.codec.Transcoder;
+import com.couchbase.client.java.kv.GetOptions;
+import com.couchbase.client.java.kv.GetResult;
+import com.couchbase.client.java.kv.MutationResult;
+import com.couchbase.client.java.kv.PersistTo;
+import com.couchbase.client.java.kv.ReplicateTo;
+import com.couchbase.client.java.kv.UpsertOptions;
+import org.apache.nifi.services.couchbase.exception.CouchbaseErrorHandler;
+import org.apache.nifi.services.couchbase.exception.CouchbaseException;
+import org.apache.nifi.services.couchbase.utils.CouchbaseGetResult;
+import org.apache.nifi.services.couchbase.utils.CouchbaseUpsertResult;
+import org.apache.nifi.services.couchbase.utils.DocumentType;
+import org.apache.nifi.services.couchbase.utils.JsonValidator;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Predicate;
+
+import static java.util.Map.entry;
+import static org.apache.nifi.services.couchbase.exception.CouchbaseErrorHandler.ErrorHandlingStrategy.FAILURE;
+import static org.apache.nifi.services.couchbase.exception.CouchbaseErrorHandler.ErrorHandlingStrategy.RETRY;
+import static org.apache.nifi.services.couchbase.exception.CouchbaseErrorHandler.ErrorHandlingStrategy.ROLLBACK;
+
+class StandardCouchbaseClient implements CouchbaseClient {
+
+    private final Collection collection;
+    private final DocumentType documentType;
+    private final PersistTo persistTo;
+    private final ReplicateTo replicateTo;
+
+    StandardCouchbaseClient(Collection collection, DocumentType documentType, PersistTo persistTo, ReplicateTo replicateTo) {
+        this.collection = collection;
+        this.documentType = documentType;
+        this.persistTo = persistTo;
+        this.replicateTo = replicateTo;
+    }
+
+    @Override
+    public CouchbaseGetResult getDocument(String documentId) throws CouchbaseException {
+        try {
+            final GetResult result = collection.get(documentId, GetOptions.getOptions().transcoder(getTranscoder(documentType)));
+
+            return new CouchbaseGetResult(result.contentAsBytes(), result.cas());
+        } catch (Exception e) {
+            throw new CouchbaseException("An error occurred while getting the document from Couchbase", e);
```

Comment:
Recommend including the `documentId` in the message
```suggestion
            throw new CouchbaseException("Failed to get document [%s] from Couchbase".formatted(documentId), e);
```